fourcc.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python3
+
+import sys
+
+fourcc_include = sys.argv[1]
+
+ident_list = []
+with open(fourcc_include) as f:
+	for l in f.readlines():
+		parts = l.split()
+		if len(parts) < 3 or parts[0] != "#define":
+			continue
+		ident = parts[1]
+		if not ident.startswith("DRM_FORMAT_") and not ident.startswith("I915_FORMAT_"):
+			continue
+		if "(" in ident:
+			# Some complicated formats use a macro with parameters
+			continue
+		ident_list.append(ident)
+
+f_name = sys.argv[2]
+
+with open(f_name, "w+") as f:
+	f.write('#include <stdint.h>\n')
+	f.write('#include <drm_fourcc.h>\n')
+	f.write('\n')
+	f.write('const char *format_str(uint32_t format)\n{\n')
+	f.write('\tswitch (format) {\n')
+	for ident in ident_list:
+		if not "_MOD_" in ident and ident !=  "DRM_FORMAT_RESERVED":
+			f.write('\t\tcase ' + ident + ':\n\t\t\treturn "' + ident[11:] + '";\n')
+	f.write('\t\tdefault:\n\t\t\treturn \"unknown\";\n')
+	f.write('\t}\n}\n\n')
+	f.write('const char *modifier_str(uint64_t modifier)\n{\n')
+	f.write('\t/*\n\t * ARM has a complex format which we can\'t be bothered to parse.\n\t */\n')
```

Comment:
Now that this is happening at build time, we could parse/generate all of the combinations for formats like this.
Although it doesn't have to be done as part of this PR; I'm just thinking out loud.